File: frontend/src/api/knowledge.ts
import { apiGet, apiPut } from "./client";

export interface Relationship {
  source_id: string;
  target_id: string;
  type: string;
  note: string | null;
}

export interface KnowledgeDetail {
  id: string;
  title: string;
  category: string;
  metadata: Record<string, unknown>;
  body: string;
  relationships: Relationship[];
}

export function fetchKnowledgeDetail(conceptId: string): Promise<KnowledgeDetail> {
  return apiGet<KnowledgeDetail>(`/knowledge/${encodeURIComponent(conceptId)}`);
}

export function updateKnowledge(
  conceptId: string,
  body: string,
  metadata: Record<string, unknown>,
): Promise<KnowledgeDetail> {
  return apiPut<KnowledgeDetail>(`/knowledge/${encodeURIComponent(conceptId)}`, { body, metadata });
}

File: backend/app/schemas/knowledge.py
from pydantic import BaseModel


class KnowledgeSummary(BaseModel):
    id: str
    title: str
    category: str


class KnowledgeListResponse(BaseModel):
    concepts: list[KnowledgeSummary]


class RelationshipOut(BaseModel):
    source_id: str
    target_id: str
    type: str
    note: str | None = None


class KnowledgeDetailResponse(BaseModel):
    id: str
    title: str
    category: str
    metadata: dict
    body: str
    relationships: list[RelationshipOut]


class UpdateKnowledgeRequest(BaseModel):
    body: str
    metadata: dict
